### PR TITLE
Add test for JSON object nesting depth

### PR DESCRIPTION
**Pull request by oshadmi**
Limit is from [serde_json](https://github.com/serde-rs/json/blob/dab5ed3ee97cef5e2b796513f8d9e4c7416e44bf/src/error.rs#L245) lib

### CODE REVIEW

**Review by MeirShpilraien — APPROVED** [review]

File: docs/docs/_index.md [hunk 1]
```diff
@@ -173,6 +173,11 @@ Lastly, you can also use the [`MODULE LOAD`](/commands/module-load) command. Not
 
 Once the module has been loaded successfully, the Redis log should have lines similar to:
 
+
+### Limitation
```
Comment (nermiller) [inline docs/docs/_index.md:177]: Should this limitation go after the log lines?

File: docs/docs/_index.md [hunk 2]
```diff
@@ -173,6 +173,11 @@ Lastly, you can also use the [`MODULE LOAD`](/commands/module-load) command. Not
...
+
+A JSON value passed to a command can have a depth of up to 128. When passing to a command a JSON value containing an object or an array with a nesting level of more than 128, then the command will return an error.
```
Comment (nermiller) [inline docs/docs/_index.md:179]:
```suggestion
A JSON value passed to a command can have a depth of up to 128. If you pass to a command a JSON value that contains an object or an array with a nesting level of more than 128, the command returns an error.
```

**Review by nermiller — CHANGES_REQUESTED** [review]
Just some minor feedback

File: tests/pytest/test.py [hunk 3]
```diff
@@ -1114,6 +1115,46 @@ def testCopyCommand(env):
     r.assertEqual(json.loads(res), [new_values])
 
 
+def nest_object(depth, max_name_len, leaf_key, leaf_val):
```
Comment (MeirShpilraien) [inline tests/pytest/test.py:1118]: Why not build it as python dict and then decode it as json?

File: tests/pytest/test.py [hunk 4]
```diff
@@ -1114,6 +1115,46 @@ def testCopyCommand(env):
...
+    res = r.execute_command('JSON.GET', 'test', '$..__leaf')
+    r.assertEqual(res, '[42]')
+
+    # No overall max nesting level (can exceeded the single value max nesting level)
+    doc = nest_object(depth, 5, "__deep_leaf", 420)
+    r.execute_command('JSON.SET', 'test', '$..__leaf', doc)
+    res = r.execute_command('JSON.GET', 'test', '$..__deep_leaf')
+    r.assertEqual(res, '[420]')
+
+    doc = nest_object(depth, 5, "__helms_deep_leaf", 42000)
```
Comment (MeirShpilraien) [inline tests/pytest/test.py:1146]: So we can create as deep nesting object as we want but not in a single `json.set` command? Is that right? Should we have some total limit as well (Maybe configurable)?

**Review by MeirShpilraien — COMMENTED** [review]
Small comments

File: tests/pytest/test.py (hunk 3 above)
Comment (oshadmi) [reply tests/pytest/test.py:1118]: Initially tried with python dict but It was throwing some exception

File: tests/pytest/test.py (hunk 4 above)
Comment (oshadmi) [reply tests/pytest/test.py:1146]: Yes, the `test` doc will exceed the limit.
The limit is coming from `serde_json` when we deserialize the JSON value argument in the command.
Why limit if we don't have to? I suppose few users, if any, would get here.
Seems enough for me to document and add a test.

File: tests/pytest/test.py (hunk 3 above)
Comment (MeirShpilraien) [reply tests/pytest/test.py:1118]: So lets understand the exception and fix them...

File: tests/pytest/test.py (hunk 4 above)
Comment (MeirShpilraien) [reply tests/pytest/test.py:1146]: Lets discuss this on another issue...

File: tests/pytest/test.py (hunk 3 above)
Comment (oshadmi) [reply tests/pytest/test.py:1117]: Fixed

**Review by MeirShpilraien — APPROVED** [review]

**Pull request merged**